### ISSUE TITLE
Accessibility: "View as Data Table Button" part of tab sequential navigation in Shadow Dom Context

### ISSUE THREAD

**Issue by adforprez** [highcharts/highcharts#22682]
#### Expected behaviour
The "View as Data Table Button" should not be part of tab sequential Navigation when using the export-data module when using HighCharts in Shadow Dom Context

#### Actual behaviour
The "View as Data Table Button" is part of tab sequential Navigation and confuses Users with ScreenReader/VoiceOver technologies.

#### Live demo with steps to reproduce
https://jsfiddle.net/w5r9vsu1/15/

- Use HighCharts with the export module, export-data module and accessibility module imported
- Place Chart into a `<figure class="highcharts-figure">` element to improve accessibility experience (https://www.highcharts.com/blog/tutorials/highcharts-keyboard-navigation-chart/)
- Place the figure element into an element with an open shadow root.

Using VoiceOver/ Screen Readers, the "View as Data Table" `<button>` element is part of the tab sequential Navigation but it should not. It should have `tabindex="-1"`

#### Reasons for bug

- The Accessibility module adds the tabindex="-1" to the button with javascript by using the getElement utility method.
initDataTableButton Method: https://github.com/highcharts/highcharts/blob/0f65d808fdfe04be2c55dd857506d6a4121a30cb/ts/Accessibility/Components/InfoRegionsComponent.ts#L793
getElement utility: 
https://github.com/highcharts/highcharts/blob/0f65d808fdfe04be2c55dd857506d6a4121a30cb/ts/Accessibility/Utils/HTMLUtilities.ts#L191

**However, the getElement utility does not penetrate shadow roots and fails to query the <button> element that requires `tabindex="-1"` to be excluded from tab sequential navigation**

![Image](https://github.com/user-attachments/assets/536e2915-baa4-463d-82bc-64ac7084f045)

#### Product version
Using HighCharts npm package, version 11.4.8, but issue persists in latest v12 version.

#### Affected browser(s)
Chrome, Firefox and Safari

**Comment by KacperMadej** [comment]
Hi @adforprez 
Could you share more info for the reason behind the behavior you have described as expected?

The current behavior can be changed using the API option [`accessibility.screenReaderSection.beforeChartFormat`](https://api.highcharts.com/highcharts/accessibility.screenReaderSection.beforeChartFormat).

**Comment by adforprez** [comment]
@KacperMadej we are not looking to modify default highcharts accessibility behavior via the https://api.highcharts.com/highcharts/accessibility.screenReaderSection.beforeChartFormat option.  
The behavior we expect is for the default `<button>` element that is added to via highCharts should not be part of tab sequential navigation (since screen readers are reading over the button and it's content when it is not tabbed on a actual data table button).

Let me know if you need any more info

**Comment by marvin19** [comment]
Hi @adforprez ,

I've tested the demo you provided and it seems to be a bug, as `tabindex` is not set on the screen reader data table button, when placed in a shadowRoot tree.

**Comment by marvin19** [comment]
Hi @adforprez ,

I’ll take a closer look and investigate in the near future. I'll keep you updated.
